build/azDevops/azure/templates/steps/build-java.yml

```diff
@@ -146,15 +167,56 @@ steps:
 
   # SonarCloud
   - ${{ if eq(parameters.static_code_analysis, true) }}:
+      - bash: |
+          set -euxo pipefail
+          function strip_refs()
+          {
+            local BRANCH_REF="$1"
+
+            RETURN_BRANCH="$(sed -e "s%^refs/\(heads\|tags\)/%%" <<< $BRANCH_REF)"
+          }
+
+          if [ -z "${{ parameters.pullrequest_number }}" ]; then
+            strip_refs "${{ parameters.source_branch_ref }}"
+            EXTRA_SONAR_ARGUMENTS="-Dsonar.branch.name='$RETURN_BRANCH'"
+            strip_refs "${{ parameters.target_branch_ref }}"
+            EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.branch.target='$RETURN_BRANCH'"
+          else
+            PROVIDER_LOWERCASED="$(tr '[:upper:]' '[:lower:]' <<< "${{ parameters.sonar_pullrequest_provider }}")"
+
+            EXTRA_SONAR_ARGUMENTS="-Dsonar.pullrequest.key='${{ parameters.pullrequest_number }}'"
+            strip_refs "${{ parameters.source_branch_ref }}"
+            EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.pullrequest.branch='$RETURN_BRANCH'"
+            EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.pullrequest.provider='$PROVIDER_LOWERCASED'"
+            strip_refs "${{ parameters.target_branch_ref }}"
+            EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} -Dsonar.pullrequest.base'$RETURN_BRANCH'"
```

Comment:
is the intention here to just overwrite the `EXTRA_SONAR_ARGUMENTS` ? or am I misreading this?

Reply:
`EXTRA_SONAR_ARGUMENTS="${EXTRA_SONAR_ARGUMENTS} ...` appends something to the end of the var

Reply:
Although I see I'm missing an `=` near the end of that line.